setting fullScopeAllowed attribute to false in a custom client results in a OptimisticLockException in the new JPA Map store hosted on an external postgres db

### Describe the bug

setting fullScopeAllowed attribute to false in a custom client results in a OptimisticLockException in the new JPA Map store hosted on an external postgres db.

The server throws the below server side exception

```
14:01:56,615 ERROR XNIO-1 task-3 [io.undertow.request] UT005023: Exception handling request to /auth/admin/realms/realm-0/clients/c9f93077-2d23-47ce-923f-315586a47a16
org.keycloak.models.ModelException: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [org.keycloak.models.map.storage.jpa.client.entity.JpaClientEntity#c9f93077-2d23-47ce-923f-315586a47a16]
	at org.keycloak.connections.jpa.PersistenceExceptionConverter.convert(PersistenceExceptionConverter.java:84)
	at org.keycloak.models.map.storage.jpa.JpaTransactionWrapper.commit(JpaTransactionWrapper.java:53)
	at org.keycloak.services.DefaultKeycloakTransactionManager.commit(DefaultKeycloakTransactionManager.java:136)
	at org.keycloak.services.filters.AbstractRequestFilter.close(AbstractRequestFilter.java:64)
	at org.keycloak.services.filters.AbstractRequestFilter.filter(AbstractRequestFilter.java:49)
	at org.keycloak.testsuite.UndertowRequestFilter.doFilter(UndertowRequestFilter.java:47)
.
.
.
.
Caused by: javax.persistence.OptimisticLockException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [org.keycloak.models.map.storage.jpa.client.entity.JpaClientEntity#c9f93077-2d23-47ce-923f-315586a47a16]
	at org.hibernate.internal.ExceptionConverterImpl.wrapStaleStateException(ExceptionConverterImpl.java:223)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:93)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181)
	at org.hibernate.internal.ExceptionConverterImpl.convertCommitException(ExceptionConverterImpl.java:65)
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:107)
	at org.keycloak.models.map.storage.jpa.JpaTransactionWrapper.commit(JpaTransactionWrapper.java:51)
	... 38 more
Caused by: org.hibernate.StaleObjectStateException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [org.keycloak.models.map.storage.jpa.client.entity.JpaClientEntity#c9f93077-2d23-47ce-923f-315586a47a16]
	at org.hibernate.persister.entity.AbstractEntityPersister.check(AbstractEntityPersister.java:2541)
	at org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3376)

```

Full Log is here: 
[optimistic_lock_exception_client_jpa_map_storage_on_psql.log](https://github.com/keycloak/keycloak/files/8618853/optimistic_lock_exception_client_jpa_map_storage_on_psql.log)



The GUI throws an `An unexpected error occurred on the server side` error on the front end causing confusing UX to the end user.

![optimisticLockException_updating_attribute_under_scope_in_a_client](https://user-images.githubusercontent.com/3415105/166648022-b3eb1e87-a58a-47d7-8f31-57c74acbb5ea.png)




### Version

main branch - SHA - acd4f5f793fb4c9eeaf135b57317fa29bce632f1

### Expected behavior

We should be able to successfully turn ON or OFF the `Full Scope Allowed` toggle for a client under a realm.

### Actual behavior

Toggling the `Full Scope Allowed` option, results in a continous optimistic lock exception in the server side and an unhandled front end level exception.

### How to Reproduce?

Bring up the new storage using the instructions from this page : https://github.com/keycloak/keycloak/discussions/9596

Choose the postgres sql option to bring up the undertow server.

Now create a new realm, under that new realm, now create a custom client with the below options.
  - create a client with the name gatling
  - set access type to confidential
  - check Service Account Enabled
  - enter a valid redirect uri (e.g. http://localhost)
  - click save
  - Change to the tab Sevice Account Roles
  - select for realm-management in the Client Roles listbox
  - assign the roles manage-clients and view-users
  - In to the Scopes tab
  - Set Full Scope Allowed to OFF (as its in ON state by default)

performing the last action triggers a front end exception as described above and a tandem server side optimistic lock exception. And the error continues to happen as we toggle the `Full Scope Allowed` to ON/OFF state

### Anything else?

_No response_

--- a/themes/src/main/resources/theme/base/admin/resources/js/controllers/clients.js
+++ b/themes/src/main/resources/theme/base/admin/resources/js/controllers/clients.js
@@ -2226,8 +2226,11 @@ module.controller('ClientScopeMappingCtrl', function($scope, $http, realm, $rout
        return $scope.client.fullScopeAllowed;
     }
 
-    $scope.changeFlag = function() {
+    $scope.changeFlag = function(event) {
         console.log('changeFlag');
+        event.stopPropagation();
+        event.preventDefault();
+        $scope.client.fullScopeAllowed = !$scope.client.fullScopeAllowed
         Client.update({
             realm : realm.realm,
             client : client.id

--- a/themes/src/main/resources/theme/base/admin/resources/partials/client-scope-mappings.html
+++ b/themes/src/main/resources/theme/base/admin/resources/partials/client-scope-mappings.html
@@ -15,7 +15,7 @@
                     <label class="col-md-2 control-label" for="fullScopeAllowed">{{:: 'full-scope-allowed' | translate}}</label>
                     <kc-tooltip>{{:: 'full-scope-allowed.tooltip' | translate}}</kc-tooltip>
                     <div class="col-md-6">
-                        <input kc-read-only="!client.access.manage" ng-model="client.fullScopeAllowed" ng-click="changeFlag()" name="fullScopeAllowed" id="fullScopeAllowed" onoffswitch on-text="{{:: 'onText' | translate}}" off-text="{{:: 'offText' | translate}}" />
+                        <input kc-read-only="!client.access.manage" ng-model="client.fullScopeAllowed" ng-click="changeFlag($event)" name="fullScopeAllowed" id="fullScopeAllowed" onoffswitch on-text="{{:: 'onText' | translate}}" off-text="{{:: 'offText' | translate}}" />
                     </div>
                 </div>
             </fieldset>

